index.js

```diff
@@ -1515,6 +1515,40 @@
     return d.valueOf() === d.valueOf() ? Just(d) : Nothing();
   });
 
+  //  requiredNonCapturingGroup :: [String] -> String
+  var requiredNonCapturingGroup = function(xs) {
+    return '(?:' + xs.join('|') + ')';
+  };
+
+  //  optionalNonCapturingGroup :: [String] -> String
+  var optionalNonCapturingGroup = function(xs) {
+    return requiredNonCapturingGroup(xs) + '?';
+  };
+
+  //  validFloatRepr :: String -> Boolean
+  var validFloatRepr = R.test(new RegExp(
+    '^' +                     // start-of-string anchor
+    '\\s*' +                  // any number of leading whitespace characters
+    '[+-]?' +                 // optional sign
+    requiredNonCapturingGroup([
+      'Infinity',             // "Infinity"
+      'NaN',                  // "NaN"
+      requiredNonCapturingGroup([
+        '[0-9]+',             // number
+        '[0-9]+[.][0-9]+',    // number with interior decimal point
+        '[0-9]+[.]',          // number with trailing decimal point
+        '[.][0-9]+'           // number with leading decimal point
+      ]) +
+      optionalNonCapturingGroup([
+        '[Ee]' +              // "E" or "e"
+        '[+-]?' +             // optional sign
+        '[0-9]+'              // exponent
+      ])
+    ]) +
+    '\\s*' +                  // any number of trailing whitespace characters
+    '$'                       // end-of-string anchor
+  ));
+
```

Comment:
This *could* be written on one line as a regexp literal:

```javascript
/^\s*[+-]?(?:Infinity|NaN|(?:\d+|\d+[.]\d*|[.]\d+)(?:[Ee][+-]?\d+)?)\s*$/
```

This is one of the rare situations in which I'd prefer to read 20 lines of code than one.